opm/simulators/wells/WellInterfaceFluidSystem.cpp

```diff
@@ -836,6 +836,17 @@ checkMaxRatioLimitCompletions(const SingleWellState& ws,
         }
     } // end of for (const auto& completion : completions_)
 
+    // The well got a ratio above the max_ratio, but none of the completions did
+    // This can happen for rare cases with very small rates.
+    // Note that the well rates are adjusted after they are accumulated up for convergence reasons
+    // and are therefore not exactly the sum of all its connection rates.
+    // For these cases we reset the violated flag for the well and continue instead of aborting
+    // due to missing worst_offending_completion
+    if (max_ratio_completion == 0) {
```

Comment:
I might read the code wrong. 

With the code above, 
``` c++
        if (ratio_completion > max_ratio_completion) {
            worst_offending_completion = completion.first;
            max_ratio_completion = ratio_completion;
        }
```

We should always find some completion here, right. Unless there is no completions for this well at all?
Under what circumstances, `max_ratio_completion == 0)`?

Reply:
I tried to explain it in the comment in the code:
```
   // The well got a ratio above the max_ratio, but none of the completions did
    // This can happen for rare cases with very small rates.
    // Note that the well rates are adjusted after they are accumulated up for convergence reasons
    // and are therefore not exactly the sum of all its connection rates.
    // For these cases we reset the violated flag for the well and continue instead of aborting
    // due to missing worst_offending_completion
```

So in a perfect world we should always find a completion but in the real world it may not due to minor differences between the sum of the perforation rates and the well rate.

Reply:
Excuse me if am too slow about here, 
`max_ratio_completion` is initialized as `0`. 

```C++
double max_ratio_completion = 0;
```

If we go through all the completions with the code below, 
```c++
        if (ratio_completion > max_ratio_completion) {
            worst_offending_completion = completion.first;
            max_ratio_completion = ratio_completion;
        }
```

If at the end `(max_ratio_completion == 0) `, 
the only way I can think about is that for all the completions, `ratio_completion <=0` or there is no completion for this well. 

Is that  (for all the completions, `ratio_completion <=0`) the situation you are looking at?

Reply:
Yes. That is correct. In particular `ratio_completion `will be zero for completions where flow is flowing out from not into the producer. In the particular case I am debugging the well has an epsilon positive production rate with a water cut above the limit (0.95) while the all the completion (only one in this case) is injecting and hence return ratio_completion =0.

Reply:
This is my suggestion. 

Use the following if statement instead, 
```c++
if (worst_offending_completion == INVALIDCOMPLETION) 
```
Then add the comment to say that `no worst offending completion is found because all the completions are injecting` due to the reason you mentioned above. 

In my opinion, this if statement can be easier to understand, and also we say clearly that the reason is because all the completions are injecting for this producer.